carma_messenger_vehicle_plugin/src/msger_veh_cfg.py

```diff
@@ -0,0 +1,171 @@
+                    }
+                    self._veh_state_dict[veh_id] = 0
+            logging.info("Vehicle configuration loaded successfully.")
+        except FileNotFoundError:
+            logging.error("Configuration file not found. Please check the file path.")
+        except json.JSONDecodeError:
+            logging.error("Error decoding JSON. Please check the input JSON file for errors.")
+        except Exception as e:
+            logging.error(f"An unexpected error occurred: {e}")
+
+    def get_veh_route(self, veh_id):
+        """
+        Retrieves the route for a specified vehicle ID.
+
+        Parameters:
+        - veh_id: The ID of the vehicle.
+        type: str
+
+        Returns:
+        - list of str: The route associated with the vehicle.
+        """
+        return self._msger_veh_dict[veh_id]["route"]
+    
+    def get_veh_speed(self, veh_id):
+        """
+        Retrieves the speed setting for a specified vehicle ID.
+
+        Parameters:
+        - veh_id: The ID of the vehicle.
+        type: str
+
+        Returns:
+        - float: The speed of the vehicle.
+        """
+        return self._msger_veh_dict[veh_id]["speed"]
+    
+    def get_veh_ids(self):
+        """
+        Retrieves a list of all vehicle IDs managed by this configuration.
+
+        Returns:
+        - list of str: A list of vehicle IDs.
+        """
+        return list(self._msger_veh_dict.keys())
+    
+    def get_veh_departure_time(self, veh_id):
+        """
+        Retrieves the departure time for a specified vehicle ID.
+
+        Parameters:
+        - veh_id: The ID of the vehicle.
+        type: str
+
+        Returns:
+        - int: The departure time of the vehicle.
+        """
+        return self._msger_veh_dict[veh_id]["departureTime"]
+    
+    def get_veh_lcm(self, veh_id):
+        """
+        Retrieves the lane change mode setting for a specified vehicle ID.
+
+        Parameters:
+        - veh_id: The ID of the vehicle.
+        type: str
+
+        Returns:
+        - int: The lane change mode of the vehicle.
+        """
+        return self._msger_veh_dict[veh_id]["lcm"]
+    
+    def get_veh_cfm(self, veh_id):
+        """
+        Retrieves the car-following model setting for a specified vehicle ID.
+
+        Parameters:
+        - veh_id: The ID of the vehicle.
+        type: str
+
+        Returns:
+        - str: The car-following model of the vehicle.
+        """
+        return self._msger_veh_dict[veh_id]["cfm"]
+    
+    def set_veh_state(self, veh_id, state):
+        """
+        Sets the state of a specified vehicle ID.
+
+        Parameters:
+        - veh_id: The ID of the vehicle.
+        - state: The new state to set for the vehicle.
+        * 0: vehicle not created
+        * 1: vehicle is created and is driving in the scene
+        * 2: vehicle is created but finish the route and destroyed
+        type: int
+        """
+        if veh_id in self._veh_state_dict:
```

Comment:
Should this be `...veh_state_dict.keys`? Or does `in` already check the key by itself? I can't recall

Reply:
Yes, `in` operator checks for the presence of a key by default.